app/src/test/java/com/example/myspacexdemoapp/ui/launch/LaunchDetailsViewModelTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.example.myspacexdemoapp.ui.launch
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Response
+import com.example.myspacexdemoapp.api.SpaceXApi
+import com.example.spacexdemoapp.GetLaunchQuery
+import io.reactivex.rxjava3.core.Observable
+import junit.framework.TestCase
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Answers
+import org.mockito.ArgumentCaptor
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class LaunchDetailsViewModelTest : TestCase() {
+
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
+    private val apolloClient: ApolloClient = mock(ApolloClient::class.java)
+    private val spaceXApi = mock(SpaceXApi(apolloClient)::class.java)
+    private val viewModel by lazy {
+        LaunchDetailsViewModel(spaceXApi)
+    }
+
+    @Test
+    fun `when launch initialized then success is retrieved`() {
```

Comment:
You have the same names for tests in `LaunchDetailsViewModelTest` and `LaunchesViewModelTest `, in some cases, it would be complicated to find what test failed.